### PR TITLE
Graph nodeattack, edgeattack, subgraphs

### PR DESCRIPTION
**Pull request by egolol**
Add the following methods to _Graph:

- [x] subgraphs
- [x] nodeattack
- [x] edgeattack
- [x] Updated LocalEfficiency measure

### CODE REVIEW

File: braph2genesis/src/gt/_Graph.gen.m [hunk 1]
```diff
@@ -726,6 +726,167 @@ function checkNegativity(negativity_type, A)
...
+
+if nargin < 3
+    layernumbers = 1:1:g.layernumber();
+end
+
+A = g.get('A');
+
+switch Graph.getGraphType(g)
+    case Graph.GRAPH
+        if iscell(A)
```
Comment (annacanal) [inline braph2genesis/src/gt/_Graph.gen.m:792]: isnt 'A' always a cell, now?

**Review by annacanal — APPROVED** [review]

File: braph2genesis/src/gt/_Graph.gen.m (hunk 1 above)
Comment (egolol) [reply braph2genesis/src/gt/_Graph.gen.m:792]: that's B no?

Comment (annacanal) [reply braph2genesis/src/gt/_Graph.gen.m:792]: No! B is a matrix for graphs, but A is a cell for all graphs. I have re-checked the graphs

Comment (egolol) [reply braph2genesis/src/gt/_Graph.gen.m:792]: could remove check then.

**Review by annacanal — APPROVED** [review]

File: braph2genesis/src/gt/_Graph.gen.m [hunk 2]
```diff
@@ -726,6 +726,165 @@ function checkNegativity(negativity_type, A)
...
 end
+% subgraphs
+function sg = subgraph(g, nodes)
+    %SUBGRAPH extracts subgraph
+    %
+    % SG = SUBGRAPH(G, NODES) extracts the graph SG as a subgraph of G
+    %  containing only the nodes specified by NODES.
+    %  If NODES is a vector, the specified nodes are removed from
+    %  all layers. If NODES is a cell array of vectors, the
+    %  specified nodes are removed from each layer.
```
Comment (giovannivolpe) [inline braph2genesis/src/gt/_Graph.gen.m:737]: not removed, but kept

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

File: braph2genesis/src/gt/_Graph.gen.m (hunk 2 above)
Comment (annacanal) [reply braph2genesis/src/gt/_Graph.gen.m]: done

File: braph2genesis/src/gt/_Graph.gen.m [hunk 3]
```diff
@@ -726,6 +726,163 @@ function checkNegativity(negativity_type, A)
...
 end
+% subgraphs
+function sg = subgraph(g, nodes)
+    %SUBGRAPH extracts subgraph
+    %
+    % SG = SUBGRAPH(G, NODES) extracts the graph SG as a subgraph of G
+    %  containing only the nodes specified by NODES.
+    %  If NODES is a vector, the specified nodes are kept from
+    %  all layers. If NODES is a cell array of vectors, the
+    %  specified nodes are removed from each layer.
```
Comment (giovannivolpe) [inline braph2genesis/src/gt/_Graph.gen.m:737]: same as before

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

**Pull request merged**